### PR TITLE
feat: Add doc for new audit service

### PR DESCRIPTION
**Pull request by vietnguyen**

### CODE REVIEW

**Review by larshelge — APPROVED** [review]

File: src/commonmark/en/content/sysadmin/audit.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
+# Audit
+
+## Introduction
+
+DHIS2 supports a new audit service which is based on Apache ActiveMQ Artemis. Artemis is used as an asynchronous messaging system by DHIS2.
+
+Aafter an entity is saved to database, an audit message will be sent to the Artemis message consumer service. The message will then be processed in a different thread.
```
Comment (luciano-fiandesio) [inline src/commonmark/en/content/sysadmin/audit.md:7]: Typo: Aafter -> After

File: src/commonmark/en/content/sysadmin/audit.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+Audit logs can be retrieved from the DHIS2 database. Currently there is no UI or API endpoint available for retriving audit entries.
+
+Note that by default, the auditing service is disabled.
```
Comment (luciano-fiandesio) [inline src/commonmark/en/content/sysadmin/audit.md:11]: This has changed, CREATE/UPDATE/DELETE auditing is enabled for all scopes now.

File: src/commonmark/en/content/sysadmin/audit.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+| uid        | text                        |               |
+| code       | text                        |               |
+
+## Setup
+
+The audit matrix configuration can be defined in `dhis.conf`  file as below. By default, audit is disabled.
+ 
+```properties
+metadata.audit.persist = on
+tracker.audit.persist = on
```
Comment (luciano-fiandesio) [inline src/commonmark/en/content/sysadmin/audit.md:37]: I think that `metadata.audit.persist` and `tracker.audit.persist` no longer exist.

File: src/commonmark/en/content/sysadmin/audit.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
+# Audit
```
Comment (luciano-fiandesio) [inline src/commonmark/en/content/sysadmin/audit.md:1]: I would add a simple description of what Auditing Scopes and Audit type.

The new Audit service makes use of two new concepts: Audit Scopes and Audit Type.

### Audit Scope

An Audit Scope is a logical area of  the application which can be audited. Currently there are three Audit Scopes:

- Tracker
- Metadata
- Aggregate

For the Tracker Audit Scope, the audited objects are: Tracked Entity Instance, Enrollment, Event, Tracked Entity Attribute Value.
For the Metadata Scope, all "metadata" objects are audited.
For the Aggregate Scope, the Aggregate Data Value objects are audited.

### Audit Type

An Audit Type is an action that triggers an audit operation. Currently we support the following types:

- READ
- CREATE
- UPDATE
- DELETE

As an example, when a new Tracked Entity Instance gets created, and if configured like so, the `CREATE` action is used to insert a new Audit entry in the `audit` db table.

File: src/commonmark/en/content/sysadmin/audit.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+All audit entries will be saved into one single table named `audit`.
+
+| Column     | Type                        | Description   |
+|------------|-----------------------------|---------------|
+| auditid    | integer                     |               |
+| audittype  | text                        | READ, CREATE, UPDATE, DELETE, SEARCH |
+| auditscope | text                        | METADATA, AGGREGATE, TRACKER |
+| klass      | text                        | Audit Entity Java class name |
+| attributes | jsonb                       | Json string stores attributes of the audit entity, used for searching. Example: {"valueType":"TEXT", "categoryCombo":"SWQW313FQY", "domainType":"TRACKER"} |   |
+| data       | bytea                       | Compressed Json string of the Audit Entity
```
Comment (luciano-fiandesio) [inline src/commonmark/en/content/sysadmin/audit.md:25]: I think we should explain how to read this field.

File: src/commonmark/en/content/sysadmin/audit.md [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+## Setup
+
+The audit matrix configuration can be defined in `dhis.conf`  file as below. By default, audit is disabled.
+ 
+```properties
+metadata.audit.persist = on
+tracker.audit.persist = on
+audit.metadata = CREATE;UPDATE;DELETE
+audit.tracker = READ;CREATE;UPDATE;DELETE
```
Comment (luciano-fiandesio) [inline src/commonmark/en/content/sysadmin/audit.md:39]: I would expand a bit on the configuration:

The audit system is automatically configured to audit for the following scopes and types:
- `CREATE`, `UPDATE`, `DELETE`
- `METADATA`, `TRACKER`, `AGGREGATE`

No action is required to activate the audit.
The audit can still be configured using the "audit matrix". The audit matrix is driven by 3 properties in `dhis.conf`:

- `audit.metadata`
- `audit.tracker`
- `audit.aggregate`

Each property accepts a semicolon delimited list of valid Audit Types:

- `CREATE`
- `UPDATE`
- `DELETE`
- `READ`

For instance, in order to only audit Tracker related object creation and deletion, the following property should be added  to `dhis.conf`:

`audit.tracker = CREATE;UPDATE`

In order to completely disable auditing, this is the configuration to use:

`audit.metadata = `
`audit.tracker = `
`audit.aggregate = `

**Pull request merged**